pkg/kudoctl/cmd/install.go

```diff
@@ -19,15 +21,53 @@ var (
 		kubectl kudo install kafka --package-version=0`
 )
 
+// parseParameters parse raw parameter strings into a map of keys and values
+func parseParameters(raw []string) (map[string]string, error) {
+	var errs []string
+	parameters := make(map[string]string)
+
+	for _, a := range raw {
+		// Using '=' as the delimiter. Split after the first delimiter to support using '=' in values
+		s := strings.SplitN(a, "=", 2)
+		if len(s) < 2 {
+			errs = append(errs, fmt.Sprintf("parameter not set: %+v", a))
+			continue
+		}
+		if s[0] == "" {
+			errs = append(errs, fmt.Sprintf("parameter name can not be empty: %+v", a))
+			continue
+		}
+		if s[1] == "" {
+			errs = append(errs, fmt.Sprintf("parameter value can not be empty: %+v", a))
+			continue
+		}
+		parameters[s[0]] = s[1]
```

Comment:
this is incorrect... it indicates that you want to add s[0 && 1] to a map... after 1+ errors... which includes `len(s) < 2` this would result in a panic prior to returning the errors.

Reply:
a test would have caught this (nudge)

Reply:
great catch.

Reply:
Nope, this works because the invalid cases have `continue` ;)
And also - there are multiple tests for this :wink:
https://github.com/kudobuilder/kudo/blob/master/pkg/kudoctl/cmd/install_test.go#L29

Reply:
https://play.golang.org/p/PP1UXyMtfo4

Reply:
the `continue` does solve my concern... nice work!

Reply:
oh even better catch!